### PR TITLE
CU/7068-7069-Dylan-PrescriptionsUnitTests

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
Prescription Unit tests

## Screenshots/Video
N/A

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
S src/screens/HealthScreen/Pharmacy/StatusDefinition/StatusDefinition.test.tsx
S src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionDetailsBanner.test.tsx
S src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionDetails.test.tsx
M src/screens/HealthScreen/Pharmacy/RefillTrackingDetails/RefillTrackingDetails.test.tsx
S src/screens/HealthScreen/Pharmacy/PrescriptionCommon/PrescriptionListItem.test.tsx
S src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNoPrescriptions.test.tsx
S src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNotAuthorized.test.tsx
S src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNoMatches.test.tsx
M src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx
S src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by theodur — CHANGES_REQUESTED** [review]
Same issues here
https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458

**Comment by Sparowhawk** [comment]
> Same issues here [#7148 (review)](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458)

updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx [hunk 1]
```diff
@@ -1,20 +1,15 @@
 import 'react-native'
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:1]: This import can be removed

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx [hunk 2]
```diff
@@ -23,17 +18,9 @@ context('RefillRequestSummary', () => {
...
-  it('initializes correctly', async () => {
-    initializeTestInstance()
-    await waitFor(() => {
-      expect(component).toBeTruthy()
-    })
-  })
-
   describe('when all request submit successfully', () => {
     it('should display successful summary', async () => {
       await waitFor(() => {
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:26]: I don't think this needs to be wrapped in a `waitFor`. Same for the other one in this file

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx [hunk 3]
```diff
@@ -80,23 +58,13 @@ context('RefillRequestSummary', () => {
...
-      // Summary
-      expect(textView[5].props.children).toEqual('Refill request summary')
-      expect(textView[6].props.children).toEqual('ALLOPURINOL 100MG TAB')
-      expect(textView[7].props.children).toEqual('Rx #: 3636691')
-
-      // Whats next should not show
-      expect(findByTypeWithText(testInstance, TextView, 'What’s next')).toBeFalsy()
+      expect(screen.getByText("We didn't get 1 refill requests")).toBeTruthy()
+      expect(screen.getByText("We're sorry. Something went wrong on our end. Try again or contact your local VA pharmacy.")).toBeTruthy()
+      expect(screen.getByText('Try again')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:63]:
```suggestion
      expect(screen.getByRole('button', { name: 'Try again' })).toBeTruthy()
```

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx [hunk 4]
```diff
@@ -116,29 +84,18 @@ context('RefillRequestSummary', () => {
...
-      expect(textView[6].props.children).toEqual('ALLOPURINOL 100MG TAB')
-      expect(textView[7].props.children).toEqual('Rx #: 3636691')
-      expect(textView[8].props.children).toEqual('AMLODIPINE BESYLATE 10MG TAB')
-      expect(textView[9].props.children).toEqual('Rx #: 3636711A')
-
-      // Whats next
-      expect(textView[10].props.children).toEqual('What’s next')
-      expect(textView[11].props.children).toEqual("We're reviewing your refill request. Once approved, the VA pharmacy will process your refill.")
-      expect(textView[12].props.children).toEqual('If you have questions about the status of your refill, contact your provider or local VA pharmacy.')
+      expect(screen.getByText('Try again')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:87]:
```suggestion
      expect(screen.getByRole('button', { name: 'Try again' })).toBeTruthy()
```

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx [hunk 5]
```diff
@@ -36,26 +28,17 @@ context('RefillScreen', () => {
...
-  it('initializes correctly', async () => {
-    await waitFor(() => {
-      initializeTestInstance()
-    })
-    expect(component).toBeTruthy()
-  })
-
   describe('no there are no refillable prescriptions', () => {
     it('should show NoRefills component', async () => {
       await waitFor(() => {
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx:36]: This `waitFor` can probably be removed, along with the other one in this file

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx [hunk 6]
```diff
@@ -67,15 +50,8 @@ context('RefillScreen', () => {
...
-
-      await waitFor(() => {
-        const button = testInstance.findByType(VAButton)
-        button.props.onPress()
-      })
-
-      const alert = testInstance.findByType(AlertBox)
-      expect(alert).toBeTruthy()
-      expect(alert.props.title).toEqual('Please select a prescription')
+      fireEvent.press(screen.getByText('Request refills'))
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx:53]:
```suggestion
      fireEvent.press(screen.getByRole('button', { name: 'Request refills' }))
```

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillTrackingDetails/RefillTrackingDetails.test.tsx [hunk 7]
```diff
@@ -1,29 +1,23 @@
 import 'react-native'
```
Comment (theodur) [inline VAMobile/src/screens/HealthScreen/Pharmacy/RefillTrackingDetails/RefillTrackingDetails.test.tsx:1]: Another import that can be removed

**Review by theodur — CHANGES_REQUESTED** [review]

File: VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionDetailsBanner.test.tsx [hunk 8]
```diff
@@ -1,30 +1,16 @@
...
-
   const initializeTestInstance = () => {
-    component = render(<PrescriptionsDetailsBanner />)
-    testInstance = component.UNSAFE_root
+    render(<PrescriptionsDetailsBanner />)
   }
 
-  it('initializes correctly', async () => {
+  it('initializes correctly', () => {
     initializeTestInstance()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionDetails/PrescriptionDetailsBanner.test.tsx:13]: updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNotAuthorized.test.tsx [hunk 9]
```diff
@@ -1,35 +1,30 @@
...
+    render(<PrescriptionHistoryNotAuthorized />)
+    expect(screen.getByText("You can't access your VA prescriptions")).toBeTruthy()
+    expect(screen.getByText('To access your VA prescriptions, upgrade your My HealtheVet account to a Premium account.')).toBeTruthy()
+    expect(screen.getByText('To upgrade, you must meet these requirements:')).toBeTruthy()
+    expect(screen.getByText("You're enrolled in VA health care, and")).toBeTruthy()
+    expect(screen.getByText("You're registered as a patient at a VA health facility.")).toBeTruthy()
+    expect(screen.getByText('Learn how to upgrade to a My HealtheVet Premium account')).toBeTruthy()
+    expect(screen.getByText('If you need help, please call the My HealtheVet help desk. We’re here Monday through Friday, 8:00 AM to 8:00 PM ET.')).toBeTruthy()
+    expect(screen.getByText('877-327-0022')).toBeTruthy()
+    expect(screen.getByText('TTY: 711')).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/PrescriptionHistory/PrescriptionHistoryNotAuthorized.test.tsx:28]: updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:1]: removed

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillTrackingDetails/RefillTrackingDetails.test.tsx (hunk 7 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillTrackingDetails/RefillTrackingDetails.test.tsx:1]: updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx (hunk 6 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx:53]: updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx (hunk 4 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:87]: updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx (hunk 3 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:63]: updated

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillRequestSummary.test.tsx:26]: removed

File: VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx (hunk 5 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/HealthScreen/Pharmacy/RefillScreens/RefillScreen.test.tsx]: updated

**Review by theodur — APPROVED** [review]

**Review by alexandec — APPROVED** [review]

**Pull request merged**